packages/target-tauri/runtime-tauri/runtime.ts

```diff
@@ -506,7 +506,16 @@ class TauriRuntime implements Runtime {
     return invoke('change_lang', { locale })
   }
   setBadgeCounter(value: number): void {
-    getCurrentWindow().setBadgeCount(value === 0 ? undefined : value)
+    const window = getCurrentWindow()
+
+    // According to the docs, `setBadgeCount` is unsupported on Windows,
+    // and we should use `setOverlayIcon` instead.
+    window.setBadgeCount(value === 0 ? undefined : value)
+    // Yes, this won't show the count.
+    window.setOverlayIcon?.(
+      value === 0 ? undefined : 'images/tray/unread-badge.png'
+    )
```

Comment:
would make sense to only run this on windows. you could pass that info to here via RuntimeInfo

Reply:
I would also prefer that. But this function is Windows-only as of now, So I suggest to keep it as is until the badge counter is supported by Tauri, or until we are otherwise forced to do it.